### DISCUSSION TITLE
Vertex position retrieval of soft bodies

### DISCUSSION THREAD

**Discussion by jankrassnigg** [jrouwe/JoltPhysics#838] [Q&A] [answered]
Happy New Year!

I am playing around with the new soft bodies, mostly for decorative purposes, such as flags and curtains.
I looked at the samples and they just use the debug rendering functionality, so they use the Get triangles API. However, I already know the triangles and connections. For rendering all I need are the shared Vertex positions, ie the particles of the soft body.
Is there a way to query those efficiently?
I couldn't find anything on the body class, and it seemed like soft bodies are not made up of shapes like rigid bodies.
I'm pretty sure I'm missing something very obvious 😄

**Comment by jrouwe** [comment, marked as answer]
If ```Body::IsSoftBody()``` then you can cast ```Body::GetMotionProperties()``` to ```SoftBodyMotionProperties``` which contains ```SoftBodyMotionProperties::GetVertices()```

**Reply by jankrassnigg** [reply]
Oh! That piece of information was the missing link, makes sense now, thanks!
